sklearn/linear_model/tests/test_ridge.py

```diff
@@ -311,6 +312,123 @@ def test_ridge_individual_penalties():
     assert_raises(ValueError, ridge.fit, X, y)
 
 
+@pytest.mark.parametrize('gcv_mode', ['svd', 'eigen'])
+@pytest.mark.parametrize('X_constructor', [np.asarray, sp.csr_matrix])
+@pytest.mark.parametrize('X_shape', [(11, 8), (11, 20)])
+@pytest.mark.parametrize('y_shape', [(11,), (11, 1), (11, 3)])
+@pytest.mark.parametrize('fit_intercept', [True, False])
+@pytest.mark.parametrize('normalize', [True, False])
+@pytest.mark.parametrize('noise', [1., 30.])
```

Comment:
Doing the complete cartesian product of all those parameters is expensive: this sequence of tests take 24s on my laptop.

May I suggest to do the cartesian product only on a subset and hard code combinations for the remaining parameters:

```diff
diff --git a/sklearn/linear_model/tests/test_ridge.py b/sklearn/linear_model/tests/test_ridge.py
index 971066265..44358a398 100644
--- a/sklearn/linear_model/tests/test_ridge.py
+++ b/sklearn/linear_model/tests/test_ridge.py
@@ -315,10 +315,14 @@ def test_ridge_individual_penalties():
 @pytest.mark.parametrize('gcv_mode', ['svd', 'eigen'])
 @pytest.mark.parametrize('X_constructor', [np.asarray, sp.csr_matrix])
 @pytest.mark.parametrize('X_shape', [(11, 8), (11, 20)])
-@pytest.mark.parametrize('y_shape', [(11,), (11, 1), (11, 3)])
-@pytest.mark.parametrize('fit_intercept', [True, False])
-@pytest.mark.parametrize('normalize', [True, False])
-@pytest.mark.parametrize('noise', [1., 30.])
+@pytest.mark.parametrize(
+    'y_shape, fit_intercept, normalize, noise',
+    [
+        ((11,), True, True, 1.),
+        ((11, 1), True, False, 20.),
+        ((11, 3), False, False, 30.),
+    ]
+)
 def test_ridge_gcv_vs_ridge_loo_cv(
         gcv_mode, X_constructor, X_shape, y_shape,
         fit_intercept, normalize, noise):
```